.github/workflows/prepare-release.yml

```diff
@@ -0,0 +1,17 @@
+name: Prepare Go Release
```

Comment:
Not specifically about this line, but would love to also document this new release process somewhere. 

Right now, our only docs for it is in your PR description.